### PR TITLE
SNOW final tweaks

### PR DESCRIPTION
**Pull request by teodosii**
- Show `external_url` for an AG that has its integration connected to ServiceNow
- Addressed some of @raphael-batte 's remarks
- Prevent SNOW creation if connection to servicenow instance fails
- Skip error handling that showed 500 or 401: Unauthorized on some calls to test_connection and instead rely purely on displaying `OK` or `Error`

### CODE REVIEW

File: grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx [hunk 1]
```diff
@@ -414,11 +419,19 @@ export const IntegrationForm = observer(
         labels: labels ? [...labels] : undefined,
       };
 
-      if (formData.integration !== 'servicenow') {
+      const isServiceNow = formData.integration === 'servicenow';
+      const isCreate = id === 'new';
```
Comment (maskin25) [inline grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx:423]: `nit` I think it's better to create vars if they are consumed more than once only or if they contans complex logic behind (a large set of consecutive OR and AND statements), in this case adding a `isCreate` var looks redundant

File: grafana-plugin/src/pages/incident/Incident.tsx [hunk 2]
```diff
@@ -278,7 +278,7 @@ class _IncidentPage extends React.Component<IncidentPageProps, IncidentPageState
     const showLinkTo = !incident.dependent_alert_groups.length && !incident.root_alert_group && !incident.resolved;
     const integrationNameWithEmojies = <Emoji text={incident.alert_receive_channel.verbal_name} />;
     const sourceLink = incident?.render_for_web?.source_link;
-    const isServiceNow = incident?.alert_receive_channel?.integration === 'servicenow';
+    const isServiceNow = Boolean(incident?.external_urls?.find((el) => el.integration_type === 'servicenow'));
```
Comment (maskin25) [inline grafana-plugin/src/pages/incident/Incident.tsx:281]: `nit` you use "servicenow" string several times, let's maybe extract this string to the const like `INTEGRATION_SERVICENOW='servicenow'` or something

**Review by maskin25 — APPROVED** [review]

File: grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx (hunk 1 above)
Comment (teodosii) [reply grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx:423]: IMO this healps with readability, much easier to read `isCreate` than `id==='new'`, we have some other places like this as well

`isNew` for example on IntegrationForm

File: grafana-plugin/src/pages/incident/Incident.tsx (hunk 2 above)
Comment (teodosii) [reply grafana-plugin/src/pages/incident/Incident.tsx:281]: Done

File: grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx (hunk 1 above)
Comment (teodosii) [reply grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx]: Done, didn't notice we already had `isNew`

**Pull request merged**